### PR TITLE
Add task list slot fill support in the sectioned task list

### LINKED ISSUE #32714: [WC 6.5 beta1] WCPay task leads to a blank page in the two variants
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The WooCommerce Payments task is not working well with the two variants:
- Clicking the WCPayments task redirects the user to a blank page
- We're not showing the legal message

### Expected behavior

- Clicking in the WCPayments task should redirect the user to the WCPayments onboarding
- We need to display the legal message:
_By setting up, you are agreeing to the [Terms of Service](https://wordpress.com/tos/)_

### Actual behavior

- Clicking the WCPayments task redirects the user to a blank page

### Steps to reproduce

1. Install WC 6.5 Beta 1 and WooCommerce Admin Test Helper 0.7.2
2. Enable usage tracking
3. Install WooCommerce Payments in the Business features step
4. Visit Tools > WCA Test Helper > Experiments and enable woocommerce_tasklist_setup_experiment_2_2022_04
5. Visit the home screen and see that the legal message is missing in the WCPaymens task, click and see the blank page
6. Visit Tools > WCA Test Helper > Experiments and enable woocommerce_tasklist_setup_experiment_1_2022_04
7. Visit the home screen and see that the legal message is missing in the WCPaymens task, click and see the blank page

### WordPress Environment

WooCommerce 6.5 Beta 1

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by louwie17**
* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

The "Get Paid with WooCommerce Payments" task that is added when you install WCPay during the onboarding wizard, was not functioning correctly in the sectioned task list, causing a blank page when clicked.

It turned out that the `<WooOnboardingTaskListItem />` slot-fill was not being used in the sectioned task list, the WooCommerce Payments task makes use of this.

Closes #32714  .

### How to test the changes in this Pull Request:

1. Checkout branch on fresh site & install the WooCommerce Admin Test Helper
2. Enable usage tracking
3. Complete OBW and install WooCommerce Payments in the Business features step
4. Visit Tools > WCA Test Helper > Experiments and enable `woocommerce_tasklist_setup_experiment_2_2022_04`
5. Visit the home screen and check if the  "Get paid with WooCommerce Payments" shows the Terms of service agreement.
6. Click "Get paid with WooCommerce Payments" task to ensure that it redirects as it should. It should also trigger the `wcadmin_tasklist_click` for the payments task, you can see this in the console by running this first `localStorage.setItem( 'debug', 'wc-admin:*' );`

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

### FOR PR REVIEWER ONLY:

* [x] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce-admin/client/tasks/reminder-bar/reminder-bar.tsx [hunk 1]
```diff
@@ -160,7 +160,7 @@ export const TasksReminderBar: React.FC< ReminderBarProps > = ( {
 
 	const tracksProps = {
 		completed: completedTasksCount,
-		is_homescreen: isHomescreen,
+		is_homescreen: !! isHomescreen,
```
Comment (louwie17) [inline plugins/woocommerce-admin/client/tasks/reminder-bar/reminder-bar.tsx:163]: This seemed to cause a TS issue for me. This enforces it to always be a boolean.

**Comment by jacob-sewell** [comment]
@louwie17 I see the terms of service and the redirect seems to be working as expected, but I'm not seeing a track for clicking the payment task.
Screen Shot 2022-04-22 at 12 58 44

**Comment by louwie17** [comment]
> @louwie17 I see the terms of service and the redirect seems to be working as expected, but I'm not seeing a track for clicking the payment task.

The screenshot of the logs you shared above, are only for tracks fired from PHP, these tracks are actually triggered within JS, so you can only see them in your browser console with using `localStorage.setItem( 'debug', 'wc-admin:*' );`

**Comment by jacob-sewell** [comment]
> The screenshot of the logs you shared above, are only for tracks fired from PHP, these tracks are actually triggered within JS, so you can only see them in your browser console with using localStorage.setItem( 'debug', 'wc-admin:*' );

I had done that, but didn't realize my console wasn't showing the level they get logged on. I see it now.

Screen Shot 2022-04-22 at 13 58 26

**Review by jacob-sewell — APPROVED** [review]
Tests well.

**Pull request merged**